e2e/stack.go

```diff
@@ -53,89 +51,139 @@ func New(
 	}
-	if err := s.startCmd(anvilCmd); err != nil {
-		return err
+	deployConfig, err := opgenesis.NewDeployConfigWithNetwork(networkName, s.deployConfigDir)
+	if err != nil {
+		return fmt.Errorf("new deploy config: %v", err)
 	}
-	env.Go(func() {
-		if err := anvilCmd.Wait(); err != nil && !errors.Is(err, ctx.Err()) {
-			s.eventListener.OnAnvilErr(fmt.Errorf("run %s: %v", anvilCmd, err))
+	deployConfig.SetDeployments(l1Deployments)
+	deployConfig.L1UseClique = false // Allows node to produce blocks without addition config. Clique is a PoA config.
+
+	// Generate a deployer key and pre-fund the account
+	deployerKey, err := crypto.GenerateKey()
+	if err != nil {
+		return fmt.Errorf("generate key: %v", err)
+	}
+
+	var auxState auxDump
+
+	l1StateJSON, err := os.ReadFile(filepath.Join(s.l1stateDumpDir, "allocs-l1.json"))
+	if err != nil {
+		// check if not found
+		if os.IsNotExist(err) {
+			return fmt.Errorf("allocs-l1.json not found - run `make setup-e2e` from project root")
+		} else {
+			return fmt.Errorf("read allocs-l1.json: %v", err)
 		}
-	})
-	// NOTE: should we set a timeout on the context? Might not be worth the complexity.
-	if !s.anvilURL.IsReachable(ctx) {
-		return nil
+	}
+	err = json.Unmarshal(l1StateJSON, &auxState)
+	if err != nil {
+		return fmt.Errorf("unmarshal l1 state: %v", err)
 	}
 
-	// Fund an account.
-	anvilRPCClient, err := rpc.DialContext(ctx, s.anvilURL.String())
+	l1state, err := auxState.ToStateDump()
 	if err != nil {
-		return fmt.Errorf("dial anvil: %v", err)
+		return fmt.Errorf("auxState to state dump: %v", err)
 	}
-	anvil := NewAnvilClient(anvilRPCClient)
-	privKey, err := crypto.GenerateKey()
+
+	l1genesis, err := opgenesis.BuildL1DeveloperGenesis(deployConfig, l1state, l1Deployments)
 	if err != nil {
-		return fmt.Errorf("generate key: %v", err)
+		return fmt.Errorf("build l1 developer genesis: %v", err)
 	}
-	if err := anvil.SetBalance(ctx, crypto.PubkeyToAddress(privKey.PublicKey), 10*oneETH); err != nil { //nolint:gomnd
-		return fmt.Errorf("set balance: %v", err)
-	}
-
-	// Deploy the OP L1 contracts.
-	forgeCmd := exec.CommandContext( //nolint:gosec
-		ctx,
-		"forge",
-		"script",
-		"--root", s.contractsRootDir,
-		"-vvv",
-		fmt.Sprintf("%s:Deploy", filepath.Join(s.contractsRootDir, "scripts", "Deploy.s.sol")),
-		"--rpc-url", s.anvilURL.String(),
-		"--broadcast",
-		"--private-key", common.Bytes2Hex(crypto.FromECDSA(privKey)),
-	)
-	if err := s.startCmd(forgeCmd); err != nil {
-		return err
+
+	l1client, l1HTTPendpoint, err := ethdevnet(ctx, s.l1BlockTime, l1genesis)
+	if err != nil {
+		return fmt.Errorf("ethdevnet: %v", err)
 	}
-	if err := forgeCmd.Wait(); err != nil {
-		return fmt.Errorf("run %s: %v", forgeCmd, err)
+
+	l1url, err := url.ParseString(l1HTTPendpoint)
+	if err != nil {
+		return fmt.Errorf("new l1 url: %v", err)
 	}
-	latestL1Block, err := anvil.BlockByNumber(ctx, nil)
+
+	// NOTE: should we set a timeout on the context? Might not be worth the complexity.
```

Comment:
Hmm yeah I feel like adding a timeout for context could probably be split out to a separate task if we want to implement that eventually. I don't think it needs to be included in this PR though